test/unit/logic/delegate.js

```diff
@@ -0,0 +1,535 @@
+				// Cannot check specific error because '' coerces to false and we get error: Username is undefined
+				expect(err).to.exist;
+				done();
+			});
+		});
+
+		it('should call callback with error if username is address like', function (done) {
+			trs.asset.delegate.username = '163137396616706346l';
+
+			delegate.verify(trs, sender, function (err) {
+				expect(err).to.equal('Username can not be a potential address');
+				done();
+			});
+		});
+
+		it('should call callback with error when username contains symbols', function (done) {
+			trs.asset.delegate.username = '^%)';
+
+			delegate.verify(trs, sender, function (err) {
+				expect(err).to.equal('Username can only contain alphanumeric characters with the exception of !@$&_.');
+				done();
+			});
+		});
+
+		it('should be okay when username contains symbols which are valid', function (done) {
+			trs.asset.delegate.username = node.randomUsername() + '!@.';
+			accountsMock.getAccount.withArgs({username: trs.asset.delegate.username}, sinon.match.any).yields(null, null);
+
+			delegate.verify(trs, sender, function (err, returnedTrs) {
+				expect(err).to.not.exist;
+				expect(returnedTrs).to.equal(returnedTrs);
+				done();
+			});
+		});
+
+		it('should call callback with error when username already exists', function (done) {
+			var expectedError = 'Error: could not connect to server: Connection refused';
+			accountsMock.getAccount.withArgs({username: node.eAccount.delegateName}, sinon.match.any).yields(expectedError);
+
+			delegate.verify(trs, sender, function (err) {
+				expect(err).to.equal(expectedError);
+				accountsMock.getAccount.verify();
+				done();
+			});
+		});
+
+		it('should call callback with error when username already exists', function (done) {
+			accountsMock.getAccount.withArgs({username: node.eAccount.delegateName}, sinon.match.any).yields(null, node.eAccount);
+
+			delegate.verify(trs, sender, function (err) {
+				expect(err).to.equal('Username already exists');
+				accountsMock.getAccount.verify();
+				done();
+			});
+		});
+
+		it('should be okay for valid transaction', function (done) {
+			accountsMock.getAccount.withArgs({username: node.eAccount.delegateName}, sinon.match.any).yields(null, null);
+
+			delegate.verify(trs, sender, function (err, returnedTrs) {
+				expect(err).to.not.exist;
+				expect(returnedTrs).to.equal(trs);
+				accountsMock.getAccount.verify();
+				done();
+			});
+		});
+	});
+
+	describe.skip('process', function () {
+
+		it('should call the callback', function (done) {
+			delegate.process(trs, sender, done);
+		});
+	});
+
+	describe.skip('getBytes', function () {
+
+		it('should return null when username is empty', function () {
+			delete trs.asset.delegate.username;
+
+			expect(delegate.getBytes(trs)).to.eql(null);
+		});
+
+		it('should return bytes for signature asset', function () {
+			var delegateBytes = delegate.getBytes(trs);
+			expect(delegateBytes.toString()).to.equal(trs.asset.delegate.username);
+		});
```

Comment:
what when `Buffer.from(trs.asset.delegate.username, 'utf8');` throws an error?

Reply:
Buffer.from is called in getBytes, and getBytes internally is always called after objectNormalize has been applied to the transaction. Which should've already checked that `trs.asset.delegate.username` is a string.

Reply:
so try catch is not needed then?

Reply:
I don't think we need it.